### ISSUE TITLE
[Question] Multi-Object Adaptability Testing 

### ISSUE THREAD

**Issue by WillMandil001** [NVIDIA-Omniverse/orbit#206]
Hello, 

I am performing the manipulation lift task but with different objects in each scene (still just one object in each cloned environment). However, doing so is very slow (8x slower than just using the same object for each cloned env). Is there any intuition on whether this is a reasonable speed? 

I am currently building num_envs rigid objects in a list and then randomization loops through the rigid object list resetting pose for each env. I am wondering if there is a better method or if someone has implemented something similar?

TLDR:
Is there a fast method of having different objects in each of the cloned envs?

Best,

Willow Mandil

**Comment by samibouziri** [comment]
Hey Willow,
I am facing the same problem right now. I just wanted to clarify something. do you create num_envs rigid body for each env choose randomly one of the objects, put it in front of the robot and throw the other out of the robots reach ?

**Comment by WillMandil001** [comment]
Hello @samibouziri

At first I did that (creating num_envs number of objects and then putting all but one object in the robots workspace) but it was super slow!

So now I have created a list of rigid objects, and each one is assigned to only one of the cloned environments. This method works better but is still 8-10x slower than using a single object for all envs... 

I have now moved onto restarting the sim with one repeated object (the standard method), training for x episodes, then shutting the sim down and restarting with a different object, as this has got to be more efficient than the previous two attempts. This method is so hacky though I hate it!

Let me know if you have any better ideas...

**Comment by samibouziri** [comment]
Thanks for the quick answer.

I ended up to the same conclusion as you. Didn't find any better solution for the moment.
I was about to implement it and was wondering if we can restart the sim without closing isaac sim and if we could also restart the sim from the python script. Do you have any idea about that or did you find that anywhere in the documentation?

**Comment by Mayankm96** [comment]
Please check the updates on #186 . We have a working example shared there.

I will close this issue to avoid duplicacy.
